### PR TITLE
Add EventStore interface - remove Adapters

### PR DESCRIPTION
**Pull request by prolic**
- some tests need to be updated

### CODE REVIEW

**Comment by codeliner** [comment]
@prolic Wow! great work. it looks fantastic.

**Comment by basz** [comment]
Looks good to me!

**Comment by prolic** [comment]
Ready for final review
/cc @codeliner

**Comment by basz** [comment]
nice, looks ok to me

File: examples/quickstart.php [hunk 1]
```diff
@@ -10,29 +10,40 @@
...
  * you write a wrapper for the event dispatcher used
  * by your web framework.
  */
-$eventStore = new \Prooph\EventStore\EventStore(
-    new \Prooph\EventStore\Adapter\InMemoryAdapter(),
-    new \Prooph\Common\Event\ProophActionEventEmitter()
-);
+$eventEmitter = new ProophActionEventEmitter([
+    ActionEventEmitterAware::EVENT_APPEND_TO,
+    ActionEventEmitterAware::EVENT_APPEND_TO,
```
Comment (codeliner) [inline examples/quickstart.php]: duplicate event name

File: examples/quickstart.php [hunk 2]
```diff
@@ -10,29 +10,40 @@
...
+use Prooph\EventStore\Stream;
+use Prooph\EventStore\StreamName;
+
 /**
- * The event store has two dependencies:
- *
- * - Prooph\EventStore\Adapter\Adapter
- * - Prooph\Common\Event\ActionEventEmitter
- *
  * Here we use the InMemoryAdapter but in a real project
```
Comment (codeliner) [inline examples/quickstart.php]: s/InMemoryAdapter/InMemoryEventStore

File: examples/quickstart.php [hunk 3]
```diff
@@ -73,25 +76,20 @@
...
-
-/**
  * Next step would be to commit the transaction.
  * But let's attach a plugin first that prints some information about currently added events.
  * Plugins are simple event listeners. See the docs of prooph/common for more details about event listeners.
  */
 $eventStore->getActionEventEmitter()->attachListener(
-    'commit.post', //Most of the event store methods provide pre and post hooks
-    function (\Prooph\Common\Event\ActionEvent $actionEvent): void {
+    ActionEventEmitterAware::EVENT_APPEND_TO, // InMemoryAdapter provides event hooks
```
Comment (codeliner) [inline examples/quickstart.php]: s/InMemoryAdapter/InMemoryEventStore

File: examples/quickstart.php [hunk 4]
```diff
@@ -73,25 +76,20 @@
...
+    function (ActionEvent $actionEvent): void {
         /**
          * In the *commit.post* action event a plugin has access to
          * all recorded events which were added in the current committed transaction.
          * It is the ideal place to attach a domain event dispatcher.
          * We only use a closure here to print the recorded events in the terminal
          */
-        $recordedEvents = $actionEvent->getParam('recordedEvents');
+        $recordedEvents = $actionEvent->getParam('streamEvents');
```
Comment (codeliner) [inline examples/quickstart.php]: We should collect and list all BC breaks, also the small ones like changed action event param names.

**Comment by malukenho** [comment]
@prolic nice work! It looks nice

**Comment by codeliner** [comment]
@prolic great work! Just found minor things.

File: examples/quickstart.php (hunk 3 above)
Comment (prolic) [inline examples/quickstart.php]: done

File: examples/quickstart.php (hunk 1 above)
Comment (prolic) [inline examples/quickstart.php]: cleaned up

File: examples/quickstart.php (hunk 2 above)
Comment (prolic) [inline examples/quickstart.php]: done

**Comment by prolic** [comment]
cleaned up

**Comment by prolic** [comment]
Build hangs on travis, because they have troubles currently. But last build before was green, also green locally. Should be good to merge.

File: src/InMemoryEventStore.php [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    }
+
+    public function appendTo(StreamName $streamName, Iterator $streamEvents): void
+    {
+        $argv = ['streamName' => $streamName, 'streamEvents' => $streamEvents];
+
+        $event = $this->actionEventEmitter->getNewActionEvent(self::EVENT_APPEND_TO, $this, $argv);
+
+        $this->getActionEventEmitter()->dispatch($event);
+
```
Comment (sandrokeil) [inline src/InMemoryEventStore.php]: @prolic Any reasons why `$this->actionEventEmitter` and `$this->getActionEventEmitter()` is used here? We should use `$this->actionEventEmitter`.

Comment (prolic) [inline src/InMemoryEventStore.php]: done via https://github.com/prooph/event-store/commit/1026b561ade49d6ca4b4f0da05df9462e387f318

**Pull request merged**